### PR TITLE
Experimental: Unix Perm add support for SyncManager's upload

### PR DESCRIPTION
**Pull request by N-o-Z**
https://github.com/treeverse/dev/issues/357

### CODE REVIEW

File: pkg/local/sync.go [hunk 1]
```diff
@@ -304,18 +305,35 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 	metadata := map[string]string{
 		ClientMtimeMetadataKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
 	}
-	reader := fileWrapper{
+
+	reader := b.Reader(f)
+	if s.includePerm {
+		if strings.HasSuffix(path, uri.PathSeparator) { // Create a 0 byte reader for directories
```
Comment (itaigilo) [inline pkg/local/sync.go:311]: Why not to create an empty reader for dirs even when `includePerm == false` - 
Mainly to prevent future bugs?

Comment (itaigilo) [inline pkg/local/sync.go:311]: (Nit, I guess:) I think that extracting this block into a function (`wrapReaderWithPerms()`?) would've make this more readable.

**Review by itaigilo — APPROVED** [review]
LGTM - 
Check out the comments.

File: pkg/local/unix_permissions.go [hunk 2]
```diff
@@ -78,3 +82,15 @@ func getUnixPermissionFromStats(stats apigen.ObjectStats) (*UnixPermissions, err
 
 	return &permissions, nil
 }
+
+func getPermissions(info os.FileInfo) (*UnixPermissions, error) {
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/local/unix_permissions.go:86]: Done

File: pkg/local/sync.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/local/sync.go:311]: 1. Yes
2. I want to isolate the code that we'll need to remove later as much as possible

Comment (N-o-Z) [reply pkg/local/sync.go:311]: Usually I would agree, but see my previous comment

**Review by guy-har — APPROVED** [review]
LGTM

**Pull request merged**